maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/InputTextRouteHandler.swift

```diff
@@ -7,39 +7,48 @@ class InputTextRouteHandler : RouteHandler {
     
     func handle(request: FlyingFox.HTTPRequest) async throws -> FlyingFox.HTTPResponse {
-        guard let appId = request.query["appId"] else {
-            logger.error("Requested view hierarchy for an invalid appId")
-            return HTTPResponse(statusCode: HTTPStatusCode.badRequest)
-        }
-        
         let decoder = JSONDecoder()
         
         guard let requestBody = try? decoder.decode(InputTextRequest.self, from: request.body) else {
             let errorData = handleError(message: "incorrect request body provided")
             return HTTPResponse(statusCode: HTTPStatusCode.badRequest, body: errorData)
         }
-        
-        let response = await MainActor.run {
-            let xcuiApplication = XCUIApplication(bundleIdentifier: appId)
-            
-            logger.info("Finding element with keyboard focus")
-            
-            let element = xcuiApplication
-                .descendants(matching: .any)
-                .element(matching: NSPredicate(format: "hasKeyboardFocus == true"))
-        
-            if (!element.exists) {
-                return HTTPResponse(statusCode: .notFound)
-            }
-            
-            element.typeText(requestBody.text)
-            
-            return HTTPResponse(statusCode: .ok)
+
+        return try await withCheckedThrowingContinuation { continuation in
+
+            let ObjC = ObjCRuntime()
+            let daemonSession = ObjC.XCTRunnerDaemonSession.sharedSession()
+            let proxy = daemonSession.daemonProxy().handle as! NSObject
+
+            // ObjCRuntime does not work correctly because
+            // class_getInstanceMethod(object_getClass(proxy), "_XCT_sendString:maximumFrequency:completion:") returns null
+            // while proxy.method(for: NSSelectorFromString("_XCT_sendString:maximumFrequency:completion:")) does work
+            //
+            //            let proxy = daemonSession.daemonProxy()
+            //            proxy._XCT_sendString(requestBody.text, maximumFrequency: 1, completion: { (error: Error?) in
+            //                if let error = error {
+            //                    continuation.resume(with: .failure(error))
+            //                } else {
+            //                    continuation.resume(with: .success(HTTPResponse(statusCode: .ok)))
+            //                }
+            //            })
+
+            typealias sendStringMethod = @convention(c) (NSObject, Selector, NSString, Int, @escaping (Error?) -> ()) -> ()
+
+            let selector = NSSelectorFromString("_XCT_sendString:maximumFrequency:completion:")
+            let methodIMP = proxy.method(for: selector)
+
+            let method = unsafeBitCast(methodIMP, to: sendStringMethod.self)
+            method(proxy, selector, requestBody.text as NSString, 1, { error in
+                if let error = error {
```

Comment:
Do we want to log here if we see any errors? Can we use `logger` to log it so that it gets written to the `xctest_runner_logs`

Reply:
Awesome thanks @berikv, don't forget to update binary when you update driver code